### PR TITLE
Resend invitation

### PR DESCRIPTION
**Pull request by slawosz**
Resends invitation email and extends token.

### CODE REVIEW

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on review app 1765, works fine as expected. There's one edge case which I will raise a separate ticket as it appears to be some security issue if we fix. so we need wider discussion.

File: cosmetics-web/config/routes.rb [hunk 1]
```diff
@@ -99,6 +99,7 @@
       resources :team_members, controller: "responsible_persons/team_members", only: %i[index new create] do
         member do
           get "new-account", action: :new_account
+          get "resend-invitation", action: :resend_inviation
```
Comment (scruti) [inline cosmetics-web/config/routes.rb:102]:
```suggestion
          get "resend-invitation", action: :resend_invitation
```

File: cosmetics-web/app/controllers/responsible_persons/team_members_controller.rb [hunk 2]
```diff
@@ -43,6 +43,17 @@ def join
     redirect_to root_path
   end
 
+  def resend_inviation
```
Comment (scruti) [inline cosmetics-web/app/controllers/responsible_persons/team_members_controller.rb:46]: Typo:
```suggestion
  def resend_invitation
```

File: cosmetics-web/spec/factories/pending_responsible_person_users.rb [hunk 3]
```diff
@@ -2,14 +2,5 @@
...
     association :responsible_person, factory: :responsible_person
-    after(:create) do |pending_responsible_person_user|
-      pending_responsible_person_user.expires_at = 2.days.from_now
-    end
-
-    factory :expired_pending_responsible_person_user do
-      after(:create) do |pending_responsible_person_user|
-        pending_responsible_person_user.expires_at = 4.days.ago
-      end
-    end
```
Comment (scruti) [inline cosmetics-web/spec/factories/pending_responsible_person_users.rb:13]: Good cleaning :bow:

File: cosmetics-web/spec/requests/responsible_persons/team_members_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    context "when invitation does not belongs to responsible person" do
+      let(:invitation) { create(:pending_responsible_person_user, responsible_person: other_responsible_person) }
+
+      it "returns 404" do
+        get resend_invitation_responsible_person_team_member_path(responsible_person, invitation)
+        expect(response.status).to eq(404)
+      end
+    end
+  end
+end
```
Comment (scruti) [inline cosmetics-web/spec/requests/responsible_persons/team_members_spec.rb:37]: :clap:

**Review by scruti — APPROVED** [review]
LGTM (but please fix typo before merging :P )

**Pull request merged**